src/Tes/Repository/TesTaskPostgreSqlRepository.cs

```diff
@@ -21,30 +24,21 @@ public class TesTaskPostgreSqlRepository : IRepository<TesTask>
     {
         private readonly Func<TesDbContext> createDbContext;
         private readonly ICache<TesTask> cache;
+        private readonly ILogger logger;
 
         /// <summary>
         /// Default constructor that also will create the schema if it does not exist
         /// </summary>
         /// <param name="connectionString">The PostgreSql connection string</param>
-        public TesTaskPostgreSqlRepository(string connectionString, ICache<TesTask> cache)
-        {
-            this.cache = cache;
-            createDbContext = () => { return new TesDbContext(connectionString); };
-            using var dbContext = createDbContext();
-            dbContext.Database.MigrateAsync().Wait();
-        }
-
-        /// <summary>
-        /// Default constructor that also will create the schema if it does not exist
-        /// </summary>
-        /// <param name="connectionString">The PostgreSql connection string</param>
-        public TesTaskPostgreSqlRepository(IOptions<PostgreSqlOptions> options, ICache<TesTask> cache)
+        public TesTaskPostgreSqlRepository(IOptions<PostgreSqlOptions> options, ILogger<TesTaskPostgreSqlRepository> logger, ICache<TesTask> cache = null)
         {
             this.cache = cache;
+            this.logger = logger;
             var connectionString = new ConnectionStringUtility().GetPostgresConnectionString(options);
             createDbContext = () => { return new TesDbContext(connectionString); };
             using var dbContext = createDbContext();
             dbContext.Database.MigrateAsync().Wait();
+            WarmCacheAsync().Wait();
```

Comment:
Not sure this needs to be a blocking operation at start up.  Requests received before the cache is consistent are okay to go the DB the first time. 

Also, there's a service that starts and runs only once in TES and happens on the background - this could be handy if you reconsider the blocking operation.

Reply:
Fortunately warming the cache takes less than 2 seconds.  There is a possible race condition if we don't block (need to analyze how likely) if the BatchScheduler updates a TES task to `COMPLETED` _before_ the cache warming is complete, because it will result in the TesTask still in memory with state `RUNNING` and Cromwell never thinking the task is complete until it expires from the cache (30 days) or TES is restarted.

Reply:
I'll implement the above in another PR